### PR TITLE
Add boot firmware package support for QCS9100

### PR DESCRIPTION
**Pull request by adhudase**
Add the boot firmware package to install boot binaries for the QCS9100 based boards.

While at it, consolidate common code into firmware-qcom-boot-common.inc to avoid repetition.

### CODE REVIEW

File: recipes-bsp/firmware/firmware-qcom-boot-qcs9100_00006.0.bb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+SRC_URI = "https://${FW_ARTIFACTORY}/${FW_BUILD_ID}/${FW_BIN_PATH}/${BOOTBINARIES}.zip"
+SRC_URI[sha256sum] = "480682759e27d63b0e44501ae2517b3671bea6dad21071880a22ed5feb5a458b"
+
+S = "${WORKDIR}/sources"
+UNPACKDIR = "${S}"
+
+INHIBIT_DEFAULT_DEPS = "1"
+
+do_configure[noexec] = "1"
+do_compile[noexec] = "1"
```
Comment (lumag) [inline recipes-bsp/firmware/firmware-qcom-boot-qcs9100_00006.0.bb:19]: I think few previous strings should go to `firmware-qcom-boot-common.inc`, they are repetitive between recipes.

Comment (ricardosalveti) [reply recipes-bsp/firmware/firmware-qcom-boot-qcs9100_00006.0.bb:19]: Yeah, @adhudase try to move common pieces to firmware-qcom-boot-common.inc.

Comment (adhudase) [reply recipes-bsp/firmware/firmware-qcom-boot-qcs9100_00006.0.bb]: Done, Thanks.

**Review by ndechesne — APPROVED** [review]

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**